python/pyspark/sql/group.py

```diff
@@ -194,6 +194,37 @@ def pivot(self, pivot_col, values=None):
             jgd = self._jgd.pivot(pivot_col, values)
         return GroupedData(jgd, self.sql_ctx)
 
+    def apply(self, udf_obj):
+        """
+        Maps each group of the current [[DataFrame]] using a pandas udf and returns the result
+        as a :class:`DataFrame`.
+
+        """
+        from pyspark.sql.functions import pandas_udf
+
+        if not udf_obj._vectorized:
+            raise ValueError("Must pass a pandas_udf")
```

Comment:
Maybe reword to "The argument to apply() must be a pandas_udf"

Reply:
Changed.